cmake/OpenCVDetectWebNN.cmake

```diff
@@ -0,0 +1,35 @@
+ocv_clear_vars(HAVE_WEBNN)
+ocv_clear_vars(WEBNN_EMSDK)
+if(NOT EMSCRIPTEN)
+  if(WITH_WEBNN)
+    set(WEBNN_HEADER_DIRS "$ENV{WEBNN_NATIVE_DIR}/gen/src/include")
+    set(WEBNN_INCLUDE_DIRS "$ENV{WEBNN_NATIVE_DIR}/../../src/include")
+    set(WEBNN_LIBRARIES "$ENV{WEBNN_NATIVE_DIR}/libwebnn_native.so;$ENV{WEBNN_NATIVE_DIR}/libwebnn_proc.so")
```

Comment:
> `$ENV{WEBNN_NATIVE_DIR}`

It is better to use CMake variable instead (you can still initialize this from environment, see usage of `ocv_check_environment_variables`).